tests/e2e/util/common/e2e_utils.go

```diff
@@ -314,61 +333,30 @@ func CheckPodsReady(ctx context.Context, cl client.Client, namespace string) err
 	return nil
 }
 
-// GetSampleYAML returns the URL of the yaml file for the testing app.
-// args:
-// version: the version of the Istio to get the yaml file from.
-// appName: the name of the testing app. Example: helloworld, sleep, tcp-echo.
-func GetSampleYAML(version istioversion.VersionInfo, appName string) string {
-	// This func will be used to get URLs for the yaml files of the testing apps. Example: helloworld, sleep, tcp-echo.
-	// Default values points to upstream Istio sample yaml files. Custom paths can be provided using environment variables.
-
-	// Define environment variables for specific apps
-	envVarMap := map[string]string{
-		"tcp-echo-dual-stack": "TCP_ECHO_DUAL_STACK_YAML_PATH",
-		"tcp-echo-ipv4":       "TCP_ECHO_IPV4_YAML_PATH",
-		"tcp-echo":            "TCP_ECHO_IPV4_YAML_PATH",
-		"tcp-echo-ipv6":       "TCP_ECHO_IPV6_YAML_PATH",
-		"sleep":               "SLEEP_YAML_PATH",
-		"helloworld":          "HELLOWORLD_YAML_PATH",
-		"sample":              "HELLOWORLD_YAML_PATH",
-		"httpbin":             "HTTPBIN_YAML_PATH",
-	}
-
-	// Check if there's a custom path for the given appName
-	if envVar, exists := envVarMap[appName]; exists {
-		customPath := os.Getenv(envVar)
-		if customPath != "" {
-			return customPath
-		}
+// GetKustomizeDir returns the path to the Kustomize directory for a test application.
+// The path is determined with the following priority:
+// 1. App-specific environment variable (e.g., HTTPBIN_KUSTOMIZE_PATH).
+// 2. Custom base path defined in CUSTOM_SAMPLES_PATH.
+// 3. Default path within the project in this case will be: `tests/e2e/samples/httpbin“.
+func GetKustomizeDir(appName string) string {
+	config, exists := appConfigs[appName]
+	if !exists {
+		return "" // return empty string if appName is not configured
 	}
 
-	// Default paths if no custom path is provided
-	var path string
-	switch appName {
-	case "tcp-echo-dual-stack":
-		path = "samples/tcp-echo/tcp-echo-dual-stack.yaml"
-	case "tcp-echo-ipv4", "tcp-echo":
-		path = "samples/tcp-echo/tcp-echo-ipv4.yaml"
-	case "tcp-echo-ipv6":
-		path = "samples/tcp-echo/tcp-echo-ipv6.yaml"
-	case "sleep":
-		path = "samples/sleep/sleep.yaml"
-	case "helloworld", "sample":
-		path = "samples/helloworld/helloworld.yaml"
-	case "httpbin":
-		path = "samples/httpbin/httpbin.yaml"
-	default:
-		return ""
+	// If app specific environment variable is set, use it.
+	if customPath := os.Getenv(config.envVar); customPath != "" {
```

Comment:
```suggestion
	if customPath := os.Getenv(strings.ToUpper(strings.ReplaceAll(appName, "-", "_") + "_KUSTOMIZE_PATH")); customPath != "" {
```

Since the "aliases" don't seem to be used, I think the map can be omitted altogether, and you can use the app name as the base of the env var name).